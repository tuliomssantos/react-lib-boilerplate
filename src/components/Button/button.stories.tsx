import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import Button from '.'

type ComponentType = typeof Button

export default {
  title: 'Components/Button',
  component: Button,
} as ComponentMeta<ComponentType>

// Create a master template for mapping args to render the Button component
const Template: ComponentStory<ComponentType> = args => <Button {...args} />

// Reuse that template for creating different stories
export const Primary = Template.bind({})
Primary.args = { children: 'Primary ðŸ˜ƒ', size: 'large' }

export const Secondary = Template.bind({})
Secondary.args = { ...Primary.args, primary: false, children: 'Secondary ðŸ˜‡' }
