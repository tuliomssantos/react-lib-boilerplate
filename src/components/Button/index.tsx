import clsx from 'clsx'
import { forwardRef } from 'react'

import styles from './styles.module.scss'

export interface ButtonProps extends React.ComponentProps<'button'> {
  backgroundColor?: string
  children: React.ReactNode
  size?: 'small' | 'medium' | 'large'
  variant?: 'primary' | 'secondary'
}

const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      backgroundColor = '',
      children,
      className,
      size = 'medium',
      variant = 'primary',
      ...rest
    },
    ref
  ) => {
    return (
      <button
        {...rest}
        ref={ref}
        type="button"
        className={clsx([
          styles.btn,
          styles[`btn__${variant}`],
          styles[`btn__${size}`],
          className,
        ])}
        style={{ backgroundColor }}
      >
        {children}
      </button>
    )
  }
)

Button.displayName = 'Button'

export default Button
