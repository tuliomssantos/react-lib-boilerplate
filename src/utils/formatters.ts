export const upperCase = (word: string) => word.toUpperCase()
