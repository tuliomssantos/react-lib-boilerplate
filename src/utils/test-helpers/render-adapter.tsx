import React from 'react'

import '@testing-library/jest-dom'

import { render, RenderResult } from '@testing-library/react'

global.React = React

export const renderAdapter = (ui: React.ReactElement): RenderResult => {
  return render(ui)
}
