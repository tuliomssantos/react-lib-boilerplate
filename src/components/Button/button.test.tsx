import { renderAdapter } from '../../utils/test-helpers'

import Button from '.'

describe('<Button />', () => {
  test('should render correctly', async () => {
    const { container } = renderAdapter(<Button>Simple button</Button>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
