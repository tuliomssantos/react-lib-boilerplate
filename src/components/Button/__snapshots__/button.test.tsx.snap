// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<button
  class="btn btn__primary btn__medium"
  type="button"
>
  Simple button
</button>
`;
